modules/entity_version_history/tests/src/FunctionalJavascript/HistoryTabSettingsTest.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Drupal\Tests\entity_version_history\Functional;
+
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\Tests\SchemaCheckTestTrait;
+
+/**
+ * Ensures the entity version history config is correctly saved.
+ *
+ * @group entity_version_history
+ */
+class HistoryTabSettingsTest extends WebDriverTestBase {
+
+  use SchemaCheckTestTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'entity_version',
+    'entity_version_history',
+    'entity_version_history_test',
+  ];
+
+  /**
+   * A user with administrative permissions.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create user.
+    $this->adminUser = $this->drupalCreateUser([
+      'access entity version history configuration',
+      'access administration pages',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
+
+  /**
+   * Tests whether the history settings form is correctly saving the settings.
+   */
+  public function testHistoryTabSettingsForm() {
+    $this->drupalGet('admin/config/entity-version/history-tab');
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+
+    // Check we have the entity type checkboxes.
+    $node_entity_checkbox = $assert_session->elementExists('css', '#edit-entity-types-node');
+    $history_entity_checkbox = $assert_session->elementExists('css', '#edit-entity-types-history-entity-test');
+
+    // Collect the bundle checkboxes and check that they are not visible.
+    $first_bundle_checkbox = $assert_session->elementExists('css', '#edit-node-first-bundle');
+    $second_bundle_checkbox = $assert_session->elementExists('css', '#edit-node-second-bundle');
+    $history_bundle_checkbox = $assert_session->elementExists('css', '#edit-history-entity-test-history-entity-test');
+
+    $this->assertFalse($first_bundle_checkbox->isVisible());
+    $this->assertFalse($second_bundle_checkbox->isVisible());
+    $this->assertFalse($history_bundle_checkbox->isVisible());
+
+    // Check the entity type checkboxes.
+    $node_entity_checkbox->check();
+    $history_entity_checkbox->check();
```

Comment:
You are checking both of these so we won't know that `$node_entity_checkbox` actually controls `$first_bundle_checkbox` and `$second_bundle_checkbox` and NOT `$history_bundle_checkbox`.